lib/SIL/SILConstants.cpp

```diff
@@ -418,6 +426,61 @@ ArrayRef<SymbolicValue> SymbolicValue::getAggregateValue() const {
   return value.aggregate->getElements();
 }
 
+//===----------------------------------------------------------------------===//
+// Enums
+//===----------------------------------------------------------------------===//
+
+namespace swift {
+
+/// This is the representation of a constant enum value. It maintains
+/// an optional SymbolicValue as a trailing object.
+struct alignas(SymbolicValue) EnumSymbolicValue final
```

Comment:
Once this is the representation that is only used for the "has a payload" case, you can simplify it to not use TrailingObjects: it can just store the (now required!) SymbolicValue inline.

Reply:
SG. It seems this way we won't need bump point allocator at all for enums.